### DISCUSSION TITLE
"Freeze current selection" — create arbitrary selections without requiring regex searches

### DISCUSSION THREAD

**Discussion by Axlefublr** [helix-editor/helix#11923] [Ideas]
I'm thinking of a feature to possibly implement into my [fork](https://github.com/Axlefublr/helix) (I don't see it getting accepted into core), and want to get feedback on the idea. Currently I don't really have a clue on to how to write it, even.

Commonly, I find myself in a situation where I have made a selection; then I want to go and create another selection arbitrarily.

For a more clear example:

```lua
function something()
	-- this is a comment
	local nevermind = 'value'
end
```

Imagine that I want to swap `something` and `nevermind` here. I can't use `C` because of the comment, and while I can make a search from select mode, to create another selection, I now have to type in the word while I'd much rather just travel to it normally.

What I want to get is the ability to "freeze" a selection. I press some mapping and that *freezes* the current selection, creating another cursor. I move the secondary cursor, but no matter what I do, the frozen one doesn't move.

This way, I can create selections however I want manually, and freeze them one by one, until I get all the selections I want. Then I press the hotkey for the "unfreeze" action and do editing actions with all the selections I now acquired.

How would this even work? Is this reasonably addable to the helix codebase, or does it depend too strongly on selections not being frozen? Would massively appreciate insight into this.

**Comment by kirawi** [comment]
https://github.com/helix-editor/helix/issues/703
